app/constants/post_draft.js

```diff
@@ -1,11 +1,12 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-export const MAX_FILE_COUNT = 5;
-export const IS_REACTION_REGEX = /(^\+:([^:\s]*):)$/i;
-export const INSERT_TO_DRAFT = 'insert_to_draft';
-export const INSERT_TO_COMMENT = 'insert_to_comment';
-export const ICON_SIZE = 24;
 export const ACCESSORIES_CONTAINER_NATIVE_ID = 'channelAccessoriesContainer';
 export const CHANNEL_POST_TEXTBOX_CURSOR_CHANGE = 'onChannelTextBoxCursorChange';
-export const CHANNEL_POST_TEXTBOX_VALUE_CHANGE = 'onChannelTextBoxValueChange';
\ No newline at end of file
+export const CHANNEL_POST_TEXTBOX_VALUE_CHANGE = 'onChannelTextBoxValueChange';
+export const ICON_SIZE = 24;
+export const INSERT_TO_COMMENT = 'insert_to_comment';
+export const INSERT_TO_DRAFT = 'insert_to_draft';
+export const IS_REACTION_REGEX = /(^\+:([^:\s]*):)$/i;
+export const MAX_FILE_COUNT = 5;
+export const MAX_MESSAGE_LENGTH_FALLBACK = 4000;
```

Comment:
this is not entirely true... there is a config value on the MaxLength thou

Reply:
True, but as of today, `MaxPostSize` is not settable via mobile/web/redux, as best I could tell. No System Console setting either, as best I can tell.

Please confirm or refute this assertion below:

The default `16383` value [comes from the database table for posts being defined as type `TEXT`](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/post_store.go#L1541-L1562) and doesn't get changed anywhere else along the way to a client, nor within a client.

I saw the `4000` value being used in a few places, but only if `Config.MaxPostSize` was never getting set somehow to the server default. So, it looked like a failsafe value, and I figured it should be labelled as so. Can change it back if my analysis is flawed.

Reply:
Share Extension has a limit of 4000 since it doesn't get max post size from `Config` , and so was not refactored (left alone).

Reply:
@lieut-data can you chip in here?

Reply:
@chuttam I know we have this https://github.com/mattermost/mattermost-mobile/blob/94bdd8e4301e845191deaa434e322dcdaf26db1d/app/components/post_draft/index.js#L84 and https://github.com/mattermost/mattermost-mobile/blob/d4c682bdddfbe82eece7b6bdf53c0f55fad0da85/ios/UploadAttachments/UploadAttachments/StoreManager.m#L200

Reply:
> @chuttam I know we have this
> 

Yup... with MaxPostSize not being set by any client, it retains the server-sent default value (`16383`). `MAX_MESSAGE_LENGTH` is never used.

Reply:
I'm not entirely sure of the ask here, but I'm curious why we need a client-side fallback at all? Can we just default to `0` until the config is loaded and available? Otherwise `4000` remains a reasonable fallback default for when we "don't know."

Reply:
I agree. I decided to leave it in exactly for the "we don't know" scenario you mention.

Once it was clear that the `4000` was merely a client-side fallback, I renamed `MAX_MESSAGE_LENGTH` to `MAX_MESSAGE_LENGTH_FALLBACK` as part of this PR, to make its purpose clearer.